code/game/objects/items/weapons/cigs.dm

```diff
@@ -101,116 +154,111 @@ LIGHTERS ARE IN LIGHTERS.DM
+/obj/item/clothing/mask/cigarette/cigarette_lighter_act(mob/living/user, mob/living/target, obj/item/direct_attackby_item)
+	var/obj/item/clothing/mask/cigarette/cig = ..()
+	if(!cig)
+		return
 
-/obj/item/clothing/mask/cigarette/attackby(obj/item/I, mob/user, params)
-	..()
-	if(istype(I, /obj/item/lighter/zippo))
-		var/obj/item/lighter/zippo/Z = I
-		if(Z.lit)
-			light("<span class='rose'>With a single flick of [user.p_their()] wrist, [user] smoothly lights [user.p_their()] [name] with [user.p_their()] [Z]. Damn [user.p_theyre()] cool.</span>")
-
-	else if(istype(I, /obj/item/lighter))
-		var/obj/item/lighter/L = I
-		if(L.lit)
-			light("<span class='notice'>After some fiddling, [user] manages to light [user.p_their()] [name] with [L].</span>")
-
-	else if(istype(I, /obj/item/match/unathi))
-		var/obj/item/match/unathi/U = I
-		if(U.lit)
-			playsound(user.loc, 'sound/effects/unathiignite.ogg', 40, FALSE)
-			light("<span class='rose'>[user] spits fire at [user.p_their()] [name], igniting it.</span>")
-			U.matchburnout()
-
-	else if(istype(I, /obj/item/match))
-		var/obj/item/match/M = I
-		if(M.lit)
-			light("<span class='notice'>[user] lights [user.p_their()] [name] with [user.p_their()] [M].</span>")
-
-	else if(istype(I, /obj/item/melee/energy/sword/saber))
-		var/obj/item/melee/energy/sword/saber/S = I
-		if(S.active)
-			light("<span class='warning'>[user] makes a violent slashing motion, barely missing [user.p_their()] nose as light flashes. [user.p_they(TRUE)] light[user.p_s()] [user.p_their()] [name] with [S] in the process.</span>")
-
-	else if(istype(I, /obj/item/assembly/igniter))
-		light("<span class='notice'>[user] fiddles with [I], and manages to light [user.p_their()] [name].</span>")
-
-	else if(istype(I, /obj/item/gun/magic/wand/fireball))
-		var/obj/item/gun/magic/wand/fireball/F = I
-		if(F.charges)
-			if(prob(50) || user.mind.assigned_role == "Wizard")
-				light("<span class='notice'>Holy shit, did [user] just manage to light [user.p_their()] [name] with [F], with only moderate eyebrow singing?</span>")
-			else
-				to_chat(user, "<span class='warning'>Unsure which end of the wand is which, [user] fails to light [name] with [F].</span>")
-				explosion(user.loc, -1, 0, 2, 3, 0, flame_range = 2)
-			F.charges--
+	if(!lit)
+		to_chat(user, "<span class='warning'>You cannot light [cig] with [src] because you need a lighter to light [src] before you can use [src] as a lighter to light [cig]... This seems a little convoluted.</span>")
+		return TRUE
 
-	//can't think of any other way to update the overlays :<
-	user.update_inv_wear_mask()
-	user.update_inv_l_hand()
-	user.update_inv_r_hand()
+	if(target == user)
+		user.visible_message(
+			"<span class='notice'>[user] presses [src] against [cig] until it lights. Seems a bit redundant...</span>",
+			"<span class='notice'>You press [src] against [cig] until it lights. Seems a bit redundant...</span>"
```

Comment:
Redundant?

Reply:
Maybe recursive would be a better joke here.

Reply:
Changed.